CMakeLists.txt

```diff
@@ -15,54 +15,91 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
-find_package(GDAL 3 REQUIRED)
 find_package(ament_cmake REQUIRED)
+find_package(grid_map_msgs REQUIRED)
 find_package(grid_map_ros REQUIRED)
 find_package(grid_map_core REQUIRED)
+find_package(rclcpp REQUIRED)
+
 find_package(Eigen3 REQUIRED)
+find_package(GDAL 3 REQUIRED)
 
-add_library(${PROJECT_NAME})
-add_subdirectory(src)
-
-target_include_directories(${PROJECT_NAME}
-    PUBLIC
-    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
-)
+find_package(yaml_cpp_vendor REQUIRED)
+link_directories(${yaml_cpp_vendor_LIBRARY_DIRS})
 
 # Reverse compatability for GDAL<3.5
 # https://gdal.org/development/cmake.html
 if(NOT TARGET GDAL::GDAL)
-    add_library(GDAL::GDAL ALIAS ${GDAL_LIBRARY})
+  add_library(GDAL::GDAL ALIAS ${GDAL_LIBRARY})
 endif()
 
-target_link_libraries(${PROJECT_NAME} Eigen3::Eigen GDAL::GDAL grid_map_core::grid_map_core)
+# Build
+add_library(${PROJECT_NAME}
+  src/grid_map_geo.cpp
+)
+
+target_include_directories(${PROJECT_NAME}
+  PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+)
 
-ament_export_targets(${PROJECT_NAME}Targets HAS_LIBRARY_TARGET)
-ament_export_dependencies(GDAL)
+target_link_libraries(${PROJECT_NAME} Eigen3::Eigen GDAL::GDAL)
+
+ament_target_dependencies(${PROJECT_NAME} SYSTEM
```

Comment:
We should use PUBLIC over SYSTEM since these aren't SYSTEM dependencies.

Reply:
Updated in [bad3291](https://github.com/ethz-asl/grid_map_geo/pull/26/commits/bad3291bbabcaf71f7a0d209b30b979b76bf59cc)